clang/test/Driver/sycl-offload-intelfpga.cpp

```diff
@@ -29,6 +29,22 @@
 // CHK-RANGE-ROUNDING: clang{{.*}} "-fsycl-is-device"{{.*}} "-fsycl-disable-range-rounding"
 // CHK-RANGE-ROUNDING: clang{{.*}} "-fsycl-disable-range-rounding"{{.*}} "-fsycl-is-host"
 
+/// FPGA target implies -emit-non-kernel-entry-points=0 in sycl-post-link
+// RUN:   %clangxx -### -target x86_64-unknown-linux-gnu -fsycl -fintelfpga %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=CHK-NON-KERNEL-ENTRY-POINTS %s
+// RUN:   %clangxx -### -target x86_64-unknown-linux-gnu -fsycl -fsycl-targets=spir64_fpga-unknown-unknown %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=CHK-NON-KERNEL-ENTRY-POINTS %s
+// CHK-NON-KERNEL-ENTRY-POINTS: sycl-post-link{{.*}} "-emit-non-kernel-entry-points=0"
+
+/// Non-FPGA targets should not imply -emit-non-kernel-entry-points=0 in sycl-post-link
+// RUN:   %clang -### -fsycl -fsycl-targets=spir64_fpga,spir64_gen %s 2>&1 \
+// RUN:   | FileCheck %s --check-prefix=CHK-NON-KERNEL-ENTRY-POINTS-NEG-1
+// CHK-NON-KERNEL-ENTRY-POINTS-NEG-1: sycl-post-link{{.*}} "-emit-non-kernel-entry-points=0"
+// CHK-NON-KERNEL-ENTRY-POINTS-NEG-1: sycl-post-link
+// CHK-NON-KERNEL-ENTRY-POINTS-NEG-1-NOT: "-emit-non-kernel-entry-points=0"
```

Comment:
Would the `-NOT` directive be checked starting from the next **line** or right after the matched substring? In all honesty, I can't find a clear explanation in the FileCheck documentation, but maybe this could be determined via an experiment of some sort.

Reply:
From [`--match-full-lines`](https://llvm.org/docs/CommandGuide/FileCheck.html#cmdoption-filecheck-match-full-lines) argument description:
> By default, FileCheck allows matches of anywhere on a line

My reading of that is that it will search for `sycl-post-link` and then start checking for `-NOT` directive right after `sycl-post-link` location in the input file.
To support my understanding, I've ran the following micro-test:
```
$ cat t.txt
sycl-post-link -emit
sycl-post-link -emit

CHECK: sycl-post-link -emit
CHECK: sycl-post-link
CHECK-NOT: -emit
$ FileCheck t.txt < t.txt
t.txt:6:12: error: CHECK-NOT: excluded string found in input
CHECK-NOT: -emit
           ^
<stdin>:2:16: note: found here
sycl-post-link -emit
               ^~~~~

Input file: <stdin>
Check file: t.txt

-dump-input=help explains the following input dump.

Input was:
<<<<<<
       1: sycl-post-link -emit
       2: sycl-post-link -emit
not:6                    !~~~~  error: no match expected
       3:
       4: CHECK: sycl-post-link -emit
       5: CHECK: sycl-post-link
       6: CHECK-NOT: -emit
>>>>>>
```

Reply:
Got it, many thanks!